Sources/Instrumentation/URLSession/URLSessionInstrumentation.swift

```diff
@@ -0,0 +1,518 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import OpenTelemetryApi
+import OpenTelemetrySdk
+
+struct NetworkRequestState {
+    var request: URLRequest?
+    var dataProcessed: Data?
+}
+
+private var idKey: Void?
+
+public class URLSessionInstrumentation {
+    private var requestMap = [String: NetworkRequestState]()
+
+    var configuration: URLSessionInstrumentationConfiguration
+
+    private let queue = DispatchQueue(label: "com.datadoghq.ddnetworkinstrumentation")
+
+    static var instrumentedKey = "com.datadoghq.instrumentedCall"
```

Comment:
question: Is there any need for our URL instrumentation to include references to DataDog? While not wrong, I feel like this could lead to confusion down the line?

Reply:
Absolutely not, I just forgot to rename them after moving this code from the implementation I was using previously.